solver/src/oneinch_solver.rs

```diff
@@ -62,26 +96,28 @@ impl OneInchSolver {
             .call()
             .await?;
 
-        let swap = self
-            .client
-            .get_swap(SwapQuery {
-                from_token_address: order.sell_token,
-                to_token_address: order.buy_token,
-                amount: order.sell_amount,
-                from_address: self.settlement_contract.address(),
-                slippage: Slippage::basis_points(MAX_SLIPPAGE_BPS).unwrap(),
-                // Disable balance/allowance checks, as the settlement contract
-                // does not hold balances to traded tokens.
-                disable_estimate: Some(true),
-                // Use at most 2 connector tokens
-                complexity_level: Some(Amount::new(2).unwrap()),
-                // Cap swap gas to 750K.
-                gas_limit: Some(750_000),
-                // Use only 3 main route for cheaper trades.
-                main_route_parts: Some(Amount::new(3).unwrap()),
-                parts: Some(Amount::new(3).unwrap()),
-            })
-            .await?;
+        let protocols = self.supported_protocols().await?;
```

Comment:
Maybe query this outside settleOrder (since we are already running into API timeout issues and this is only required once per batch).